mux_client_impl/src/builder.rs

```diff
@@ -0,0 +1,316 @@
+use super::{Error, Result, Session};
+
+use std::borrow::Cow;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+use std::str;
+
+use tempfile::Builder;
+
+use tokio::process;
+
+/// Build a [`Session`] with options.
+#[derive(Debug, Clone)]
+pub struct SessionBuilder {
```

Comment:
It should be possible to share the session builder between the two implementations — after all, they all need the same information to ultimately do SSH.

Reply:
Yep, all functions in SessionBuilder except for just_connect can be shared between them.
That is doable.